hal/src/nRF52840/button_hal.c

```diff
@@ -49,14 +39,19 @@ button_config_t HAL_Buttons[] = {
     }
 };
 
+static struct {
```

Comment:
Let's make this `volatile` just in case, as it will be accessed from two different ISRs.

Reply:
Added